src/js/common/parser.js

```diff
@@ -0,0 +1,223 @@
+    var promise = fluid.promise();
+    parser.dereference(pathOrUri, that.options.parserOptions, gpii.schema.parser.getParserCallback(that, schemaKey, promise));
+    return promise;
+};
+
+gpii.schema.parser.getParserCallback = function (that, schemaKey, promise) {
+    return function (error, schema) {
+        if (error) {
+            promise.reject(error);
+        }
+        else {
+            that.dereferencedSchemas[schemaKey] = schema;
+            promise.resolve(schema);
+        }
+    };
+};
+
+/*
+
+ Given the path to a field and a schema key, look up the definition in the JSON Schema.
+ Accepts the following input parameters when launched through the invoker:
+
+ 1. `schemaKey`: The `key` of the schema (generally the filename, i.e. `base.json`).
+ 2. `schemaFieldPath`: a string like `.root.nested.value.description` or an array of path segments like
+    `["root", "nested", "value", "description"]`.
+
+ This function assumes it is working with an object that directly represents a JSON Schema, and that child properties
+ will be represented within the `properties` element of each level.  The path `.root.nested.value.description` implies
+ a structure like the following:
+
+    {
+      root: {
+        properties: {
+          nested: {
+            properties: {
+              value: {
+                description: "Text description"
+              }
+            }
+          }
+        }
+      }
+    }
+
+  At all but the last level, the segment (`nested`, for example`) is looked for first as a property, then as a root
+  element.  On the last level, the segment (`description`, for example) is looked for first as a root element, and then
+  as a property. This is done to avoid unexpected behavior in dealing with schema properties that match JSON Schema
+  keywords, such as `required`, `title`, or `description`.
+
+ */
+
+gpii.schema.parser.lookupField = function (that, schemaKey, schemaFieldPath) {
+    var pathSegments = Array.isArray(schemaFieldPath) ? schemaFieldPath : gpii.schema.validator.extractPathSegments(schemaFieldPath);
+
+    var dereferencedSchema = that.dereferencedSchemas[schemaKey];
+    if (dereferencedSchema) {
+        var currentLevel = dereferencedSchema;
+        for (var a = 0; a < pathSegments.length; a++) {
+            var segment = pathSegments[a];
+            var nextLevel = currentLevel.properties && currentLevel.properties[segment] ? currentLevel.properties[segment] : currentLevel[segment];
+            if (a === pathSegments.length - 1) {
+                nextLevel = currentLevel[segment] ? currentLevel[segment] : nextLevel;
+            }
+            if (nextLevel) {
+                currentLevel = nextLevel;
+            }
+            else {
+                return false;
+            }
+        }
+
+        return currentLevel;
+    }
+
+    // If we can't evolve the output, return `false` so that we can use the existing error.
```

Comment:
I update the comment to clarify that `false` is returned if the field cannot be looked up, and that it is the responsibility of code that uses this function to respond to take action when a field cannot be looked up.